app/models/katello/upstream_pool.rb

```diff
@@ -19,31 +24,52 @@ def response_to_pools(response)
       end
 
       def map_attributes(pool)
-        {
-          pool_id: pool['id'],
-          active: pool['activeSubscription'],
-          quantity: pool['quantity'],
-          start_date: pool['startDate'],
-          end_date: pool['endDate'],
-          contract_number: pool['contractNumber'],
-          consumed: pool['consumed'],
-          product_name: pool['productName'],
-          product_id: pool['productId'],
-          subscription_id: pool['subscriptionId']
+        attributes = {}
+        kat_to_cp_map = {
+          pool_id: 'id',
+          active: 'activeSubscription',
+          quantity: 'quantity',
+          start_date: 'startDate',
+          end_date: 'endDate',
+          contract_number: 'contractNumber',
+          consumed: 'consumed',
+          product_name: 'productName',
+          product_id: 'productId',
+          subscription_id: 'subscriptionId'
         }
+
+        kat_to_cp_map.map do |kat, cp|
+          attributes[kat] = pool[cp] if pool[cp]
+        end
+
+        attributes
+      end
+
+      def minimal_fields
+        ['id', 'quantity']
+      end
+
+      def all_fields
```

Comment:
the candlepin field names are duplicated in this method and in `map_attributes`. we can dry it up by making the above `kat_to_cp_map` a method returning that hash and it can be used throughout this class let me know if that made any sense. Maybe it's an improvement, maybe it isn't.

minimal_fields would become something like:
```
def minimal_fields
 kat_to_cp_map.values_at(:id, :quantity)
end
```

it's more text, but the CP field names only exist in one place